File: VB/SnapExample.Module/DatabaseUpdate/Updater.vb
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Data.Filtering

Namespace SnapExampleXPO.Module.DatabaseUpdate
    Public Class Updater
        Inherits ModuleUpdater

        Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
            MyBase.New(objectSpace, currentDBVersion)
        End Sub
        Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
            MyBase.UpdateDatabaseAfterUpdateSchema()
            CreatePerson("Mary", "Tellitson")
            CreatePerson("John", "Nilsen")
        End Sub
        Private Sub CreatePerson(ByVal firstName As String, ByVal lastName As String)
            Dim person As Person = ObjectSpace.FindObject(Of Person)(CriteriaOperator.Parse("FirstName = ? and LastName = ?", firstName, lastName))
            If person Is Nothing Then
                person = ObjectSpace.CreateObject(Of Person)()
                person.FirstName = firstName
                person.LastName = lastName
            End If
        End Sub
    End Class
End Namespace

File: CS/SnapExample.Module/DatabaseUpdate/Updater.cs
using System;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Updating;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Data.Filtering;

namespace SnapExampleXPO.Module.DatabaseUpdate {
    public class Updater : ModuleUpdater {
        public Updater(IObjectSpace objectSpace, Version currentDBVersion) :
            base(objectSpace, currentDBVersion) {
        }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            CreatePerson("Mary", "Tellitson");
            CreatePerson("John", "Nilsen");
        }
        private void CreatePerson(string firstName, string lastName) {
            Person person = ObjectSpace.FindObject<Person>(CriteriaOperator.Parse("FirstName = ? and LastName = ?", firstName, lastName));
            if (person == null) {
                person = ObjectSpace.CreateObject<Person>();
                person.FirstName = firstName;
                person.LastName = lastName;
            }
        }
    }
}
